src/Catalog.GraphQL/Product/ProductMutationType.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Catalog.Abstractions;
+using Catalog.GraphQL.Models;
+
+namespace Catalog.GraphQL.Product;
+
+[ExtendObjectType("Mutation")]
+public class ProductMutationType
+{
+    
+    public async Task<ProductDto> AddProductAsync(ProductDto productDto,
+        [Service] IApplicationDbContext context,
+        CancellationToken cancellationToken)
+    {
+        var product = ProductDto.FromProductDto(productDto);
```

Comment:
Is it possible to make it generic? So, currently the same logic is duplicated for catalogs and for the products

Reply:
Second thought: this GraphQL is just a client, as an API. So, shouldn't it use Catalog.Application layer for all logic.

Reply:
- Yep, it's possible to use generics.
- Yep, we could use logic from Catalog.Application layer. But in this case, we must rely on types that use this layer. Is it ok?

Reply:
Up to you to decide which approach to select. Some additional notes:

- If you like to go with generics, the question is how to make it extensible. Because, in a real word it's not only crud operations, so, we need to have at least one example, where there would be some additional logic.
- If you like to go with the Clean Architecture, where GraphQL is just a UI later, it's totally fine to rely on the models from the Application layer. (this is ok for API, so, why not for GraphQL, just a different UI implementation). At the level of UI, we might have additional models, it calls ViewModel. e.g. when you are building a Razor application and need a model that accumulates different information, it might be useful.

Reply:
Ok. Currently mutation level uses logic and types from Application layer. Also, these types are exposed as GrapQL types.